pcsx2/Frontend/CommonHost.cpp

```diff
@@ -382,6 +387,8 @@ void CommonHost::OnGameChanged(const std::string& disc_path, const std::string&
 #ifdef ENABLE_DISCORD_PRESENCE
 	UpdateDiscordPresence(GetRichPresenceString());
 #endif
+
+	ReloadPINE();
```

Comment:
Shouldn't this be `s_pine_server.Deinitialize()`? (don't rely on the destructor to close things down, the order that globals construct/destruct is undefined)

Reply:
I didn't really understand this comment, but I've reworked the mechanics in ReloadPINE to make sure Deinitialize is always called, see comment below

Reply:
Will that actually close things down though? If `m_end` isn't set, I don't see how it would.

When the process exits, the destructors for all global objects get executed, but the other in which they execute is unspecified, so if you have dependencies between objects, that can cause issues (e.g. B destructs before A, but A is assuming B is still valid).

AFAICT the easiest way to ensure it closes at this point, rather than in an unspecified order, would be to call Deinitialize().

Reply:
Right, I just don't understand why it should be in OnGameChanged, maybe in onShutdown?

Reply:
Should be in `CPUThreadShutdown()`, yes (GH showed the wrong "title"..)

Reply:
fixed